source/renderer/app/containers/wallet/dialogs/settings/WalletRecoveryPhraseStep2Container.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import React, { Component } from 'react';
+import { observer, inject } from 'mobx-react';
+import WalletRecoveryPhraseStep2Dialog from '../../../../components/wallet/settings/WalletRecoveryPhraseStep2Dialog';
+import WalletRecoveryPhraseStep3Dialog from '../../../../components/wallet/settings/WalletRecoveryPhraseStep3Dialog';
+import WalletRecoveryPhraseStep4Dialog from '../../../../components/wallet/settings/WalletRecoveryPhraseStep4Dialog';
+import validWords from '../../../../../../common/crypto/valid-words.en';
+import type { InjectedDialogContainerProps } from '../../../../types/injectedPropsType';
+
+type Props = InjectedDialogContainerProps;
+
+@inject('stores', 'actions')
+@observer
+export default class WalletRecoveryPhraseStep2Container extends Component<Props> {
+  static defaultProps = {
+    actions: null,
+    stores: null,
+    children: null,
+    onClose: () => {},
+  };
+
+  componentWillReceiveProps(nextProps: Props) {
+    const { walletBackup, wallets } = nextProps.stores;
+    const { actions } = this.props;
+    const {
+      isRecoveryPhraseMatching: nextRecoveryPhraseMatching,
+      getWalletIdAndBalanceRequest: getWalletIdAndBalanceRequestNext,
+    } = walletBackup;
+
+    let dialog;
+    if (getWalletIdAndBalanceRequestNext.wasExecuted) {
+      if (nextRecoveryPhraseMatching) {
+        dialog = WalletRecoveryPhraseStep3Dialog;
+        const activeWallet = wallets.active;
+        if (activeWallet) activeWallet.updateWalletLocalData();
+      } else {
+        dialog = WalletRecoveryPhraseStep4Dialog;
+      }
+      actions.dialogs.open.trigger({
+        dialog,
+      });
+      actions.walletBackup.resetRecoveryPhraseCheck.trigger();
+    }
+  }
+
+  handleVerify = (recoveryPhrase: Array<string>) => {
+    this.props.actions.walletBackup.checkRecoveryPhrase.trigger({
+      recoveryPhrase,
+    });
+  };
+
+  render() {
+    const { stores } = this.props;
+    const { wallets, walletBackup } = stores;
+    const { isValidMnemonic } = wallets;
+    const { getWalletIdAndBalanceRequest } = walletBackup;
+    const { closeActiveDialog } = this.props.actions.dialogs;
+
+    const isVerifying =
```

Comment:
@daniloprates I think we need `&&` operator instead of `||` here?

Reply:
Nope, the button needs to display the ⏳for either state.